### DISCUSSION TITLE
Unwrap doesn't unwrap the write function?

### DISCUSSION THREAD

**Discussion by stickfigure** [pmndrs/jotai#2720] [Q&A]
Hi all, new to jotai, love it, but I'm a little puzzled about `unwrap`. It seems to unwrap the read operation but not the write operation? Is that deliberate? Here's a little more about my use case:

I have an `atomWithDefault` that fetches data from a server:

```
export const thingsAtom = atomWithDefault(
    async (get) => {
        const things: Thing[] = await getSomeStuffFromTheServer();
        return things;
    }
);
```

Now I want some way to create a new thing with a server call and add it to the existing list:

```
export const addThingAtom = atom(
    null,
    async (get, set, addMe: SomeThingData) => {.   
        const newThing: Thing = await addThingOnServer(addMe);

        set(thingsAtom, async (oldPromise) => {
            const old = await oldPromise;
            return Promise.resolve([...old, added]);
        });
    }
);
```

Using `unwrap(thingsAtom)` doesn't change anything. The `set()` method still works with promises. I'd love to not have to deal with promises here.

**Comment by dai-shi** [comment]
Good point. Sounds valid.
Originally `unwrap` doesn't do anything for `write`, and afterwards #1997 supports "pass through it".
The problem is that `unwrap` can't know if the passed atom is a primitive atom. I don't think there's a feasible solution.
So, currently it's by design, but feel free to experiment and share your experience.
